controlplane/rosa/controllers/rosacontrolplane_controller.go

```diff
@@ -200,13 +207,12 @@ func (r *ROSAControlPlaneReconciler) reconcileNormal(ctx context.Context, rosaSc
 		return ctrl.Result{}, nil
 	}
 
-	cluster, err := rosaClient.GetCluster()
```

Comment:
why are we changing the flow here? I don't think we should rely on status fields to inform decisions, as those could be edited/removed manually.